Update computeHashImpl() to use a span

cdumez opened the pull request:
#### ec9483a7d0e3850cee94dd22f1635397c6a1c764
<pre>
Update computeHashImpl() to use a span
<a href="https://bugs.webkit.org/show_bug.cgi?id=271215">https://bugs.webkit.org/show_bug.cgi?id=271215</a>

Reviewed by Darin Adler and Sam Weinig.

* Source/WTF/wtf/text/AtomStringImpl.cpp:
(WTF::SubstringTranslator8::hash):
(WTF::SubstringTranslator16::hash):
* Source/WTF/wtf/text/StringHash.h:
(WTF::ASCIICaseInsensitiveHash::hash):
(WTF::HashTranslatorASCIILiteral::hash):
* Source/WTF/wtf/text/StringHasher.h:
* Source/WTF/wtf/text/StringHasherInlines.h:
(WTF::StringHasher::computeHashAndMaskTop8Bits):
(WTF::StringHasher::computeLiteralHashAndMaskTop8Bits):
(WTF::StringHasher::hashWithTop8BitsMasked):
* Source/WTF/wtf/text/StringImpl.cpp:
(WTF::StringImpl::hashSlowCase const):
(WTF::StringImpl::concurrentHash const):
* Source/WTF/wtf/text/StringImpl.h:
(WTF::HashTranslatorCharBuffer::HashTranslatorCharBuffer):
(WTF::StringImpl::setHash const):
(WTF::StringImpl::assertHashIsCorrect const):
* Source/WTF/wtf/text/StringView.h:
(WTF::StringView::hash const):
* Source/WTF/wtf/text/SuperFastHash.h:
(WTF::SuperFastHash::computeHashAndMaskTop8Bits):
(WTF::SuperFastHash::computeHash):
(WTF::SuperFastHash::computeHashImpl):
* Source/WTF/wtf/text/WYHash.h:
(WTF::WYHash::computeHashAndMaskTop8Bits):
(WTF::WYHash::hash):
(WTF::WYHash::computeHashImpl):
* Source/WebCore/bindings/js/ScriptBufferSourceProvider.h:
* Source/WebCore/loader/appcache/ApplicationCacheStorage.cpp:
(WebCore::urlHostHash):
* Source/WebCore/loader/cache/CachedScript.cpp:
(WebCore::CachedScript::script):
* Source/WebCore/platform/graphics/WidthCache.h:
(WebCore::WidthCache::SmallStringKey::SmallStringKey):

Canonical link: <a href="https://commits.webkit.org/276406@main">https://commits.webkit.org/276406@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/54268cd5461dafa78824eb302263c3ef2352ed18

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/44534 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/23611 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/46982 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/47251 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/40537 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/46839 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/27677 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/21001 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/36651 "Exiting early after 500 failures. 1145 tests run. 500 failures") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/45110 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/20710 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/38331 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/17725 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/18141 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/39475 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/2581 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/37726 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/40774 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/39751 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/48808 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/43984 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/19494 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/16027 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/43555 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/20836 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/42303 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/21164 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/51166 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/6133 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/20492 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/10374 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

weinig commented on Source/WTF/wtf/text/StringHash.h:
```diff
@@ -119,7 +119,7 @@ namespace WTF {
         static unsigned hash(const UChar* data, unsigned length)
         {
-            return StringHasher::computeHashAndMaskTop8Bits<UChar, FoldCase>(data, length);
+            return StringHasher::computeHashAndMaskTop8Bits<UChar, FoldCase>(std::span { data, length });
```
Maybe an overload of `static unsigned hash(...)` that that's a `std::span`? 

(also, this function and the one below taking `const LChar* data, unsigned length` can be merged

```c++
        template<typename CharacterType>
        static unsigned hash(const CharacterType* data, unsigned length)
        {
            return StringHasher::computeHashAndMaskTop8Bits<CharacterType, FoldCase>(std::span { data, length});
        }
```
)

cdumez committed:
Update computeHashImpl() to use a span
https://bugs.webkit.org/show_bug.cgi?id=271215

Reviewed by Darin Adler and Sam Weinig.

* Source/WTF/wtf/text/AtomStringImpl.cpp:
(WTF::SubstringTranslator8::hash):
(WTF::SubstringTranslator16::hash):
* Source/WTF/wtf/text/StringHash.h:
(WTF::ASCIICaseInsensitiveHash::hash):
(WTF::HashTranslatorASCIILiteral::hash):
* Source/WTF/wtf/text/StringHasher.h:
* Source/WTF/wtf/text/StringHasherInlines.h:
(WTF::StringHasher::computeHashAndMaskTop8Bits):
(WTF::StringHasher::computeLiteralHashAndMaskTop8Bits):
(WTF::StringHasher::hashWithTop8BitsMasked):
* Source/WTF/wtf/text/StringImpl.cpp:
(WTF::StringImpl::hashSlowCase const):
(WTF::StringImpl::concurrentHash const):
* Source/WTF/wtf/text/StringImpl.h:
(WTF::HashTranslatorCharBuffer::HashTranslatorCharBuffer):
(WTF::StringImpl::setHash const):
(WTF::StringImpl::assertHashIsCorrect const):
* Source/WTF/wtf/text/StringView.h:
(WTF::StringView::hash const):
* Source/WTF/wtf/text/SuperFastHash.h:
(WTF::SuperFastHash::computeHashAndMaskTop8Bits):
(WTF::SuperFastHash::computeHash):
(WTF::SuperFastHash::computeHashImpl):
* Source/WTF/wtf/text/WYHash.h:
(WTF::WYHash::computeHashAndMaskTop8Bits):
(WTF::WYHash::hash):
(WTF::WYHash::computeHashImpl):
* Source/WebCore/bindings/js/ScriptBufferSourceProvider.h:
* Source/WebCore/loader/appcache/ApplicationCacheStorage.cpp:
(WebCore::urlHostHash):
* Source/WebCore/loader/cache/CachedScript.cpp:
(WebCore::CachedScript::script):
* Source/WebCore/platform/graphics/WidthCache.h:
(WebCore::WidthCache::SmallStringKey::SmallStringKey):

Canonical link: https://commits.webkit.org/276406@main
